### PR TITLE
feat: implement backend members read path for application

### PR DESCRIPTION
**Pull request by jaroslawlakomy-grav**
## Issue

https://gravitee.atlassian.net/browse/APIM-13465

## Description

Add portal members listing for applications with pagination and permission checks, map membership data to the Member contract (including member.id), and cover empty/populated responses in resource tests (including links = null for empty list).

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java [hunk 1]
```diff
@@ -59,4 +61,25 @@ public Member convert(ExecutionContext executionContext, MemberEntity member, Ur
...
         return memberItem;
     }
+
+    public Member convert(ExecutionContext executionContext, io.gravitee.apim.core.member.model.Member member, UriInfo uriInfo) {
+        final Member memberItem = new Member();
+
+        memberItem.setId(member.getId());
+        memberItem.setCreatedAt(member.getCreatedAt().toInstant().atOffset(ZoneOffset.UTC));
+
+        UserEntity userEntity = userService.findById(executionContext, member.getId());
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java]: ⚠️  Do not perform db operations from mapper ⚠️

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java [hunk 2]
```diff
@@ -238,4 +279,13 @@ public Response transferMemberOwnership(
         );
         return Response.noContent().build();
     }
+
+    private SearchApplicationMembersCriteria mapSearchCriteria(ApplicationMembersSearchInput input) {
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java]: why this method is not in mapper?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.application.model;
+
+import lombok.Builder;
+
+@Builder
+public record SearchApplicationMembersCriteria(Filters filters) {
+    @Builder
+    public record Filters(String displayName) {}
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java]: Why do we introduce 2nd level here?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    public String displayName() {
+        return filters != null ? filters.displayName() : null;
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java]: it could be just a field of base record

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    public SearchApplicationMembersUseCase(ApplicationQueryService applicationQueryService, MembershipQueryService membershipQueryService) {
+        this.applicationQueryService = applicationQueryService;
+        this.membershipQueryService = membershipQueryService;
+    }
+
+    public Page<Member> execute(Input input) {
+        var executionContext = input.executionContext();
+
+        var application = applicationQueryService.searchByIds(
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java]: in this context you should just use crud service and get by id

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImpl.java [hunk 6]
```diff
@@ -15,27 +15,59 @@
...
+import java.util.function.Function;
+import java.util.stream.Collectors;
 import org.springframework.context.annotation.Lazy;
 import org.springframework.stereotype.Service;
 
 @Service
 public class MembershipQueryServiceImpl implements MembershipQueryService {
 
     private final MembershipRepository membershipRepository;
+    private final RoleQueryService roleQueryService;
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImpl.java]: with this amount of dependencies, we should create a domain service to fetch members with related references

**Review by dev-marek — CHANGES_REQUESTED** [review]
some cleanup needed

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java [hunk 7]
```diff
@@ -59,4 +61,25 @@ public Member convert(ExecutionContext executionContext, MemberEntity member, Ur
         memberItem.setUpdatedAt(member.getUpdatedAt().toInstant().atOffset(ZoneOffset.UTC));
         return memberItem;
     }
+
+    public Member convert(ExecutionContext executionContext, io.gravitee.apim.core.member.model.Member member, UriInfo uriInfo) {
```
[the remark this answers is not in the document]
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java]: there is no user lookup right now (in new code), OnionArchitecture use different Member type

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java (hunk 1 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java]: removed in new convert method

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java [hunk 8]
```diff
@@ -59,4 +61,25 @@ public Member convert(ExecutionContext executionContext, MemberEntity member, Ur
...
+        memberItem.setId(member.getId());
+        memberItem.setCreatedAt(member.getCreatedAt().toInstant().atOffset(ZoneOffset.UTC));
+
+        UserEntity userEntity = userService.findById(executionContext, member.getId());
+        User memberUser = userMapper.convert(userEntity);
+        memberUser.setLinks(
+            userMapper.computeUserLinks(usersURL(uriInfo.getBaseUriBuilder(), userEntity.getId()), userEntity.getUpdatedAt())
+        );
+
+        memberItem.setUser(memberUser);
```
[the remark this answers is not in the document]
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java]: convert does not call findById now

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java (hunk 2 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java]: created decicated mapper

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java (hunk 5 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java]: swiitched to getById, and moved to domain service

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImpl.java (hunk 6 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImpl.java]: domain service created

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImpl.java [hunk 9]
```diff
@@ -143,6 +175,188 @@ public Collection<Membership> findGroupsThatUserBelongsTo(String memberId) {
...
+        try {
+            var memberships = membershipRepository.findByReferenceAndRoleId(MembershipReferenceType.APPLICATION, applicationId, null);
+
+            var roleById = findRolesById(memberships);
+            var usersById = findUsersById(memberships);
+            var groupsById = findGroupsById(memberships);
+
+            var membersByCompositeKey = new HashMap<String, Member>();
+
+            memberships.forEach(membership -> {
```
[the remark this answers is not in the document]
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImpl.java]: outdated, method remoced

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+                    Membership.builder()
+                        .id("membership-2")
+                        .referenceType(MembershipReferenceType.APPLICATION)
+                        .referenceId(APPLICATION_ID)
+                        .memberType(MembershipMemberType.USER)
+                        .memberId("user-1")
+                        .roleId("role-2")
+                        .createdAt(Date.from(Instant.parse("2020-02-01T20:22:02.00Z")))
+                        .updatedAt(Date.from(Instant.parse("2020-02-02T20:22:02.00Z")))
+                        .build()
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java]: Please create/move to fixtures. Additional wants: use UUIDs instead of "membership-2" or "role-2" strings.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImplTest.java [hunk 11]
```diff
@@ -19,6 +19,7 @@
 import static org.assertj.core.api.Assertions.catchThrowable;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.eq;
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImplTest.java]: ?

**Review by dev-marek — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class SearchApplicationMembersDomainServiceImpl implements SearchApplicationMembersDomainService {
+
+    private final MembershipRepository membershipRepository;
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java]: better use crud service / query service

**Review by dev-marek — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java (hunk 12 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java]: no dependencies to repository form DomainService

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImplTest.java (hunk 11 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/query_service/membership/MembershipQueryServiceImplTest.java]: unused import - removed

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java (hunk 10 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java]: fixtures created

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+        try {
+            var memberships = membershipRepository.findByReferenceAndRoleId(MembershipReferenceType.APPLICATION, applicationId, null);
+
+            var roleById = findRolesById(memberships);
+            var usersById = findUsersById(memberships);
+            var groupsById = findGroupsById(memberships);
+
+            var membersByCompositeKey = new HashMap<String, Member>();
+
+            memberships.forEach(membership -> {
```
[the remark this answers is not in the document]
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java]: moved to separate method

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+        if (displayName == null) {
+            return false;
+        }
+
+        return displayName.toLowerCase(Locale.ROOT).contains(searchedDisplayName.toLowerCase(Locale.ROOT));
+    }
+
+    private Page<Member> paginate(Collection<Member> members, Pageable pageable) {
+        if (pageable == null) {
+            pageable = new PageableImpl(1, Integer.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java]: local variable used

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class SearchApplicationMembersDomainServiceImpl implements SearchApplicationMembersDomainService {
```
[the remark this answers is not in the document]
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java:53]: no repository access from DomainService now

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java (hunk 3 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java]: simplified

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java (hunk 4 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/model/SearchApplicationMembersCriteria.java]: simplified

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,260 @@
...
+    }
+
+    private void mergeMembershipIntoMembersMap(
+        Membership membership,
+        Map<String, Member> membersByCompositeKey,
+        Map<String, Role> roleById,
+        Map<String, BaseUserEntity> usersById,
+        Map<String, Group> groupsById
+    ) {
+        var member = membersByCompositeKey.computeIfAbsent(memberCompositeKey(membership), key ->
```
[the remark this answers is not in the document]
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java]: fixed

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapper.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.portal.rest.mapper;
+
+import io.gravitee.apim.core.application.model.SearchApplicationMembersCriteria;
+import io.gravitee.rest.api.portal.rest.model.ApplicationMembersSearchInput;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ApplicationMembersSearchCriteriaMapper {
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapper.java]: why not mapstruct?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java (hunk 7 above)
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java]: This should be a mapper. Why is there executionContext?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+@UseCase
+public class SearchApplicationMembersUseCase {
+
+    private final SearchApplicationMembersDomainService searchApplicationMembersDomainService;
+
+    public SearchApplicationMembersUseCase(SearchApplicationMembersDomainService searchApplicationMembersDomainService) {
+        this.searchApplicationMembersDomainService = searchApplicationMembersDomainService;
+    }
+
+    public Page<Member> execute(Input input) {
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java]: In our pattern we use record `Output`.
We don't use any feature of class `Page` so we can just return `List`

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java [hunk 19]
```diff
@@ -0,0 +1,262 @@
...
+
+    private final MembershipQueryService membershipQueryService;
+    private final RoleQueryService roleQueryService;
+    private final UserCrudService userCrudService;
+    private final GroupQueryService groupQueryService;
+    private final ApplicationCrudService applicationCrudService;
+
+    @Override
+    public Page<Member> searchApplicationMembers(
+        ExecutionContext executionContext,
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java]: In 2026 we don't pass executionContext but environmentId directly

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java [hunk 20]
```diff
@@ -0,0 +1,256 @@
...
+            );
+            when(membershipQueryService.findByReference(eq(Membership.ReferenceType.APPLICATION), eq(APPLICATION_ID))).thenReturn(
+                Set.of(
+                    anApplicationMembership(MEMBERSHIP_ID_1, APPLICATION_ID, "user-1", ROLE_ID_1),
+                    anApplicationMembership(MEMBERSHIP_ID_2, APPLICATION_ID, "user-2", ROLE_ID_2)
+                )
+            );
+            when(roleQueryService.findByIds(Set.of(ROLE_ID_1, ROLE_ID_2))).thenReturn(
+                Set.of(
+                    io.gravitee.apim.core.membership.model.Role.builder()
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java]: can we update/create fixtures?

**Review by dev-marek — CHANGES_REQUESTED** [review]
In next pass: investigate what to do with two member/membership packages

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapper.java (hunk 17 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapper.java]: switched to mapstruct

Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapper.java]: but ..... there is a lot other implementations based on @Component

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java (hunk 7 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/MemberMapper.java]: leftover from recent implementation, removed

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java (hunk 19 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImpl.java]: switched to enviromentId

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java (hunk 18 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java]: Alligned with Output pattern

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.application.use_case;
+
+import io.gravitee.apim.core.UseCase;
+import io.gravitee.apim.core.application.model.SearchApplicationMembersCriteria;
+import io.gravitee.apim.core.member.model.Member;
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java:20]: instead of Member -> `io.gravitee.apim.core.membership.model.Membership`

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java (hunk 20 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/membership/SearchApplicationMembersDomainServiceImplTest.java]: added to fixtures

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java (hunk 21 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/SearchApplicationMembersUseCase.java]: refactoring done

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java [hunk 22]
```diff
@@ -67,6 +74,12 @@ public class ApplicationMembersResource extends AbstractResource {
     @Inject
     private MemberMapper memberMapper;
 
+    @Inject
+    private ApplicationMembersSearchCriteriaMapper applicationMembersSearchCriteriaMapper;
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java]: usually we use static instance (see: `ConnectionMapper#INSTANCE`

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapperTest.java [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+import static org.junit.Assert.assertNull;
+
+import io.gravitee.rest.api.portal.rest.model.ApplicationMembersSearchFilters;
+import io.gravitee.rest.api.portal.rest.model.ApplicationMembersSearchInput;
+import org.junit.Test;
+import org.mapstruct.factory.Mappers;
+
+public class ApplicationMembersSearchCriteriaMapperTest {
+
+    private final ApplicationMembersSearchCriteriaMapper mapper = Mappers.getMapper(ApplicationMembersSearchCriteriaMapper.class);
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapperTest.java]: usually `INSTANCE` field is used

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/spring/ResourceContextConfiguration.java [hunk 24]
```diff
@@ -524,6 +534,11 @@ public ApplicationMapper applicationMapper() {
         return mock(ApplicationMapper.class);
     }
 
+    @Bean
+    public ApplicationMembersSearchCriteriaMapper applicationMembersSearchCriteriaMapper() {
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/spring/ResourceContextConfiguration.java]: to be removed (we use static instance)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/AbstractResourceTest.java [hunk 25]
```diff
@@ -290,6 +291,9 @@ public abstract class AbstractResourceTest extends JerseySpringTest {
     @Autowired
     protected ApplicationMapper applicationMapper;
 
+    @Autowired
+    protected ApplicationMembersSearchCriteriaMapper applicationMembersSearchCriteriaMapper;
```
Comment (dev-marek) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/AbstractResourceTest.java]: to be removed (we use static instance)

**Review by dev-marek — APPROVED** [review]
only one small thing (about static mapper instance) and ready to go

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java (hunk 22 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationMembersResource.java]: fixed

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapperTest.java (hunk 23 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMembersSearchCriteriaMapperTest.java]: fixed

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/spring/ResourceContextConfiguration.java (hunk 24 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/spring/ResourceContextConfiguration.java]: fixed

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/AbstractResourceTest.java (hunk 25 above)
Comment (jaroslawlakomy-grav) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/AbstractResourceTest.java]: fixed
